homeassistant/components/media_player/cast.py

```diff
@@ -68,12 +68,17 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     casts = []
 
+    all_chromecasts = pychromecast.get_chromecasts()
+
     for host in hosts:
-        try:
-            casts.append(CastDevice(*host))
-            KNOWN_HOSTS.append(host)
-        except pychromecast.ChromecastConnectionError:
-            pass
+        found = [device for device in all_chromecasts
```

Comment:
Maybe add a comment (either here, or the call to `get_chromecasts()`) explaining why, as it's not obvious from reading the code.